### PR TITLE
www-apps/netbox: Version bump 2.11.1

### PR DESCRIPTION
**Pull request by anthonyryan1**
We are no longer patching upgrade.sh to remove pip.

Because upgrade.sh runs everything inside of a virtualenv, it fails
to find many of the system dependencies that aren't installed in pip.

Going forward, people can run upgrade.sh which will duplicate the
dependencies in a virtualenv, or run the commands from inside
upgrade.sh oneby one to use the system versions.

Closes: https://bugs.gentoo.org/750836
Closes: https://bugs.gentoo.org/709330
Closes: https://bugs.gentoo.org/709448
Package-Manager: Portage-3.0.13, Repoman-3.0.2
Signed-off-by: Anthony Ryan <anthonyryan1@gmail.com>

### CODE REVIEW

**Comment by scardracs** [comment]
@anthonyryan1 do you mind bump directly into 2.11.0?

**Comment by scardracs** [comment]
You may want to add https://github.com/gentoo/gentoo/pull/18011 as closed when your PR will be merged

**Comment by anthonyryan1** [comment]
I've got 2.11.0 in my overlay, but there's a couple blockers to getting it in the tree.

* We need dev-python/tablib as a new package
* dev-python/django-tables2 needs to be updated to depend on tablib (unsure if we should make this default or a [tablib] use flag).

Then we can bump 2.11.0. I can get the other pull requests done in later today.

**Comment by scardracs** [comment]
> I've got 2.11.0 in my overlay, but there's a couple blockers to getting it in the tree.
> 
>     * We need dev-python/tablib as a new package
> 
>     * dev-python/django-tables2 needs to be updated to depend on tablib (unsure if we should make this default or a [tablib] use flag).
> 
> 
> Then we can bump 2.11.0. I can get the other pull requests done in later today.

You also need to update py-gfm https://github.com/gentoo/gentoo/pull/20435

**Comment by scardracs** [comment]
List of packages required for 2.11.0: #20435 #20442

**Comment by scardracs** [comment]
@anthonyryan1 tablib is merged! You can move to 2.11 now :)

**Comment by anthonyryan1** [comment]
@scardracs can you confirm how py-gfm fits into netbox here? I installed it in a virtualenv (since that seems to be their recommended approach) and it didn't pull in gfm anywhere I could see.

Is it somehow a drop-in alternative for the dev-python/markdown used here?

**Comment by scardracs** [comment]
> @scardracs can you confirm how py-gfm fits into netbox here? I installed it in a virtualenv (since that seems to be their recommended approach) and it didn't pull in gfm anywhere I could see.
> 
> Is it somehow a drop-in alternative for the dev-python/markdown used here?

Personally I don't know because I haven't checked more than list of deps

**Comment by scardracs** [comment]
> @scardracs can you confirm how py-gfm fits into netbox here? I installed it in a virtualenv (since that seems to be their recommended approach) and it didn't pull in gfm anywhere I could see.
> 
> Is it somehow a drop-in alternative for the dev-python/markdown used here?

I thought it still required on 2.10/11 like 2.7 did

File: www-apps/netbox/netbox-2.10.8.ebuild [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Finally, if you are using webhooks, start the netbox-rqworker service.
+
+[1] https://netbox.readthedocs.io/en/stable/installation/
+[2] https://netbox.readthedocs.io/en/stable/installation/upgrading/
+"
+
+src_install() {
+	dodir /opt
+	cp -a ../${P} "${ED}"/opt
```
Comment (SoapGentoo) [inline www-apps/netbox/netbox-2.10.8.ebuild:89]: missing `|| die`

File: www-apps/netbox/netbox-2.10.8.ebuild [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	dosym ${P} /opt/netbox
+dosym ../../etc/netbox/gunicorn_config.py /opt/netbox/gunicorn_config.py
```
Comment (SoapGentoo) [inline www-apps/netbox/netbox-2.10.8.ebuild:91]: indentation

File: www-apps/netbox/netbox-2.10.8.ebuild [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+	dosym ../../../../etc/netbox/configuration.py \
+	/opt/netbox/netbox/netbox/configuration.py
```
Comment (SoapGentoo) [inline www-apps/netbox/netbox-2.10.8.ebuild:93]: indentation here too

File: www-apps/netbox/netbox-2.10.8.ebuild [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	dodir /etc/netbox
```
Comment (SoapGentoo) [inline www-apps/netbox/netbox-2.10.8.ebuild:94]: redundant with the following statement

File: www-apps/netbox/netbox-2.10.8.ebuild [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	keepdir /var/log/netbox
+	fowners -R netbox:netbox /var/log/netbox
+	readme.gentoo_create_doc
+}
+
+pkg_postinst() {
+	readme.gentoo_print_elog
+	local r
+	for r in $REPLACING_VERSIONS; do
+		if [[ $r = "2.5.10" ]]; then
```
Comment (SoapGentoo) [inline www-apps/netbox/netbox-2.10.8.ebuild:112]:
```suggestion
		if [[ ${r} == "2.5.10" ]]; then
```

**Review by SoapGentoo — CHANGES_REQUESTED** [review]

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+KEYWORDS="~amd64"
+IUSE="ldap"
+REQUIRED_USE=${PYTHON_REQUIRED_USE}
+
+RDEPEND="
+	acct-group/netbox
+	acct-user/netbox
+	${PYTHON_DEPS}
+	$(python_gen_cond_dep '
+		>=dev-python/django-3.1[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:23]: 2.11.1 requires django 3.2 https://github.com/netbox-community/netbox/blob/develop/requirements.txt

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/django-cacheops-5.1[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/django-cors-headers-3.7[${PYTHON_MULTI_USEDEP}]
+		dev-python/django-debug-toolbar[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:26]: I suggest to use `dev-python/django-debug-toolbar-3.2.1` as per requirement

Comment (scardracs) [reply www-apps/netbox/netbox-2.11.1.ebuild:26]: I have github. There is a mayor=

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/django-filter-2.4[${PYTHON_MULTI_USEDEP}]
+		dev-python/django-mptt[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:28]: We need to update our package to 0.12.0

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/django-pglocks-1.0.4[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/django-prometheus-2.1.0[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/django-rq-2.4.0[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:31]: 2.4.1

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/django-tables2-2.3.4[${PYTHON_MULTI_USEDEP}]
+		dev-python/tablib[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/django-taggit-1.3.0[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:34]: Need to be updated to 1.4.0

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/django-timezone-field-4.1[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/djangorestframework-3.12[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:36]: 3.12.4

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 12]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/drf-yasg-1.20[${PYTHON_MULTI_USEDEP},validation]
+		www-servers/gunicorn[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:38]: 20.1.0

**Comment by scardracs** [comment]
Sorry for comments: I was signing which packages should I have to made a PR

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 13]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/jinja-2.11.3[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/markdown-3.3.4[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/netaddr-0.8[${PYTHON_MULTI_USEDEP}]
+		>=dev-python/pillow-8.1[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:42]: 8.2

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 14]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/psycopg-2.8.6[${PYTHON_MULTI_USEDEP}]
+		dev-python/pycryptodome[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:44]: 3.10.1

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 15]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/pyyaml-5.4[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:45]: 5.4.1

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 16]
```diff
@@ -0,0 +1,117 @@
...
+		>=dev-python/redis-py-3.3.11[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:46]: I don't see this package as requirement

File: www-apps/netbox/netbox-2.11.1.ebuild [hunk 17]
```diff
@@ -0,0 +1,117 @@
...
+		dev-python/svgwrite[${PYTHON_MULTI_USEDEP}]
```
Comment (scardracs) [inline www-apps/netbox/netbox-2.11.1.ebuild:47]: 1.4.1

**Review by SoapGentoo — APPROVED** [review]

**Comment by SoapGentoo** [comment]
@anthonyryan1 please also fix the comments from @scardracs, this is a last-rites candidate

**Comment by scardracs** [comment]
I'm sorry @anthonyryan1 but I have closed my PR for porting packages needed for upgrade. If you want to take a look on it you could take ebuilds I have already done and try to finish all the ebuilds I have not done yet

**Comment by SoapGentoo** [comment]
Removed from tree

**Pull request closed without merging**